### PR TITLE
Deployment Manager Package

### PR DESCRIPTION
**Pull request by davidvossel**
This PR contains a refactor of both the ```make cluster-deploy``` and the setup/teardown used in ```functests/```. All the code used to bring the OCS deployment and StorageClusters online has been moved into the ```pkg/deploy-manager``` package

The goal here is to codify the OCS and StorageCluster install process as preliminary pieces for the upcoming update/upgrade/uninstall functional tests. We need to refactor the all the deployment relate logic into a reusable package and let the functional test manage both installing OCS, installing the StorageCluster, and performing lifecycle tests on the OCS installation such as update and uninstallation. 

In order to centralize the source of truth for how OCS is installed, the ```deploy/deploy-with-olm.yaml``` file is now generated from source as well using the ```pkg/deploy-manager``` package. 

## Changes
* The ```make cluster-deploy``` now utilizes the ```tools/cluster-deploy/cluster-deploy.go``` which depends on the ```pkg/deploy-manager``` package
* The ```functest/setup_teardown.go``` logic now utilizes the shared ```pkg/deploy-manager```
* ```deploy/deploy-with-olm.yaml``` is now generated from source using ```pkg/deploy-manager```. See the ```make gen-latest-deploy-yaml``` and ```make verify-latest-deploy-yaml``` make targets

### CODE REVIEW

**Comment by davidvossel** [comment]
@jarrpa @kshlm This PR should be ready for review now. I updated the description to reflect the changes. All and all, the the major changes here are just a refactoring of the functest deploy logic and golang generated deploy/deploy-with-olm.yaml.

I centralized all the deploy logic into a package which we'll use to build the install/update/uninstall related functional tests off of.

File: tools/csv-merger/csv-merger.go [hunk 1]
```diff
@@ -12,9 +12,10 @@ import (
 	"strings"
 	"text/template"
 
-	"github.com/coreos/go-semver/semver"
+	"github.com/blang/semver"
```
Comment (jarrpa) [inline tools/csv-merger/csv-merger.go]: Why this change?

File: pkg/deploy-manager/config.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	"os"
+
+	ocsv1 "github.com/openshift/ocs-operator/pkg/apis/ocs/v1"
+	olmclient "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	restclient "k8s.io/client-go/rest"
```
Comment (jarrpa) [inline pkg/deploy-manager/config.go]: Why is this imported twice?

File: pkg/deploy-manager/config.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+func init() {
+	ocsv1.SchemeBuilder.AddToScheme(scheme.Scheme)
+}
+
+// DeployManager is a util tool used by the functional tests
+type DeployManager struct {
+	olmClient      *olmclient.Clientset
+	k8sClient      *kubernetes.Clientset
+	restClient     *rest.RESTClient
```
Comment (jarrpa) [inline pkg/deploy-manager/config.go]: Maybe name this `ocsClient`?

File: tools/csv-merger/csv-merger.go (hunk 1 above)
Comment (davidvossel) [reply tools/csv-merger/csv-merger.go]: I had to update the operator-lifecycle-manager package we use because of a bug they had when I posted OperatorGroup objects.

For some reason they switched the semver package, which called csv-merger to fail to compile until I updated our code to reflect the change as well.

File: pkg/deploy-manager/config.go (hunk 2 above)
Comment (davidvossel) [reply pkg/deploy-manager/config.go]: oh, oops.

**Comment by davidvossel** [comment]
feedback addressed
